modules/nextflow/src/main/groovy/nextflow/cloud/aws/batch/S3Helper.groovy

```diff
@@ -43,6 +43,40 @@ class S3Helper {
             unset IFS
         }
         
+        nxf_retry() {
+            local max_attempts=\${ATTEMPTS-4}
+            local timeout=\${TIMEOUT-1}
+            local attempt=0
+            local exitCode=0
+            echo "Downloading" \$2
+
+            while [[ \$attempt < \$max_attempts ]]
+            do
+              if "\$@"
```

Comment:
Ummm, this must be tested with paths containing one or more blanks.

Reply:
I actually tried it previously with s3 paths containing blanks and it didn't work. Should the paths be escaped in input files?

Reply:
Just exporting the `IFS=$'\n'` same as you did in `nxf_s3_upload`, solved the problem.
I tested paths without blanks, having blank spaces in directory name, and having blank spaces in filenames, and all of them worked.

Reply:
Want to see passing the tests [this](https://github.com/nextflow-io/nextflow/blob/754da16fc4cd15dd1392f65fc1969b264b9cd949/tests/complex-names.nf) :D

Reply:
Is this the expected result?
```
[warm up] executor > awsbatch
WARN: Config setting `executor.awscli` has been deprecated -- Use instead `aws.batch.cliPath`
executor >  awsbatch (2)
[14/a9897e] process > foo [100%] 1 of 1 ✔
[e0/afd4d6] process > bar [100%] 1 of 1 ✔
Hello world!

Completed at: 03-Jun-2019 16:43:08
Duration    : 2m 31s
CPU hours   : (a few seconds)
Succeeded   : 2
```

Reply:
Looks promising, tho still failing the unit tests. Tip: run `make test` locally before pushing the commit. It will check the unit tests (tho not the integration tests)

Reply:
Oh, I missed some tests. I'll fixed them and let you know. Thanks for tip.

Reply:
I fixed all tests related to AWS Batch, and when I run `make test` all the tests pass, but for some reason `GooglePipelinesTaskHandlerTest` fails on travis. Do you have any idea what could be the problem. I run `GooglePipelinesTaskHandlerTest` locally and all of them are passing.

Reply:
Seems related to this changes https://github.com/nextflow-io/nextflow/commit/331c1bc4ba3e3afc314a880caab5382f35404b61. But weird it happens only on travis. Have you updated recently your branch with the master? I think you are missing the last commit.

Reply:
That was the problem. I didn't update it with master for a while. Travis tests are passing now.